File: samples/java/DeliveryDispatch/Server/Configuration/src/main/java/systems/zlink/samples/deliverydispatch/server/configuration/SampleLocationStore.java
package systems.zlink.samples.deliverydispatch.server.configuration;

import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions;
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore;

import java.time.Duration;

public final class SampleLocationStore {
    private SampleLocationStore() {}

    public static ZLinkRedisLocationStore create(SampleTopology topology) {
        return new ZLinkRedisLocationStore(
                new ZLinkRedisLocationOptions()
                        .setConnectionString(topology.redisEndpoint())
                        .setKeyPrefix(topology.redisKeyPrefix() + "locations:")
                        .setCommandTimeout(Duration.ofMillis(500)));
    }
}

File: samples/kotlin/DeliveryDispatch/Server/Configuration/src/main/kotlin/systems/zlink/samples/kotlin/deliverydispatch/server/configuration/SampleLocationStore.kt
package systems.zlink.samples.kotlin.deliverydispatch.server.configuration

import java.time.Duration
import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore

object SampleLocationStore {
    fun create(): ZLinkRedisLocationStore =
        ZLinkRedisLocationStore(
            ZLinkRedisLocationOptions()
                .setConnectionString(SampleTopology.RedisEndpoint)
                .setKeyPrefix(SampleTopology.RedisKeyPrefix + "locations:")
                .setCommandTimeout(Duration.ofMillis(500))
        )
}
